src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -772,6 +772,26 @@ bool layout_optimizer::deconvolution_b_fs_yx_fsv16_opt(layout const &input_layou
     return false;
 }
 
+static bool is_node_for_onednn(deconvolution_node const& node) {
+    auto prim = node.get_primitive();
+    auto input_layout = node.get_dependency(0).get_output_layout();
+    auto output_layout = node.get_output_layout();
+
+    // Onednn deconv does not support cross-precision
+    bool onednn_valid_dt = (data_type_traits::is_i8_u8(input_layout.data_type) && data_type_traits::is_i8_u8(output_layout.data_type)) ||
```

Comment:
Looks like that you can check only one data type here (and it doesn't matter input or output) due to you have extra check on equality in the line 788.

Reply:
Nice catch ;) Can we follow-up on this in next PR? we will be enabling cross-precision deconvolution. https://github.com/intel-innersource/libraries.performance.math.onednn/pull/2653